app/models/host_status/status.rb

```diff
@@ -48,6 +54,10 @@ def relevant?
 
     private
 
+    def populate_timestamp
```

Comment:
How about moving this `if` condition to the `before_validation`, and just calling `:update_timestamp` from there?

```ruby
before_validation :update_timestamp, :unless => ->(host) { host.reported_at.present? }
```